src/compiler/symbolWalker.ts

```diff
@@ -158,6 +162,10 @@ namespace ts {
             }
 
             function visitSymbol(symbol: Symbol): boolean {
+                if (cancellationToken) {

```

Comment:
I have 2 fairly ignorant questions
- Are cancellation checks expensive (I seem to recall that it's based on checking a named pipe, which sounds scary)? Is it throttled?
- How many times do we expect `visitSymbol` to be called in a "typical" operation?

I guess the overall question I'm asking is whether you're fairly sure this check isn't a performance burden.

Reply:
[it is](https://github.com/Microsoft/TypeScript/blob/54dfdd77fb222be0cf6aa16a9e962b1d2f9a8e23/src/server/lsHost.ts#L19)

Reply:
FYI: Some crude overhead measurements from back when I prototyped this: https://github.com/billti/cancellation . Definitely use the throttling mechanism. We've seen significant perf regressions when this wasn't done in prior cancellation checks.

Reply:
As Vlad said, cancellation checks are throttled.  Having said that, I don't have a sense of how expensive cancellability is.